CHANGELOG.md

```diff
@@ -7,6 +7,12 @@ and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [0.4.3]
++ add unit tests for the hydro and two pieces of wet equation
++ bump  bottom height of user requested levels by ~1mm during ray tracing to ensure interpolation works
```

Comment:
What does this mean? Does this imply a bug in Raider? You should not need to modify code outside raider to get the tests to work.

Reply:
I wouldn't call this a bug in raider so much as a manifestation of projection error. 
If user requests the bottom layer of weather model, which is -500 m, then due to the conversion of interpolation points from [WGS84 -> ECEF](https://github.com/dbekaert/RAiDER/blob/74535ed4baac1d960ec7f6f3431f9a202d16e6e0/tools/RAiDER/delay.py#L324)(for getting the ray vectors) and then from [ECEF to WGS84](https://github.com/dbekaert/RAiDER/blob/74535ed4baac1d960ec7f6f3431f9a202d16e6e0/tools/RAiDER/delay.py#L393) (for intersecting the ray vectors with delays) the bottom height layer becomes about -500.0002. (about 0.2 mm difference).

As this new value is outside of the actual weather model height we interpolate to (-500 m) it results in nan values.  For a real weather model, this results in the loss of 30 (260) nondimensional contributions to the integral for wet (hydrostatic) delays. Considering the total integral, this changes the ray length by about 100 m for hydrostatic equation (normalized difference of 0.001). 

I'm of the opinion we should include the bottom layer, which [my change](https://github.com/dbekaert/RAiDER/blob/74535ed4baac1d960ec7f6f3431f9a202d16e6e0/tools/RAiDER/delay.py#L406) does.

Reply:
1. How would we know this without this comment above?
2. This discussion and improvement really merits its own PR
3. Can you develop a test to illustrate the improvement for illustration?

Reply:
And...

You are assuming that pts z coordinate *never* exceeds 1 mm. What happens if the z coord exceeds 1 meter or even 1 km? Sounds like an unexpected side-effect. There is no reference to 1 mm in this update.

Reply:
hmm looks like i slightly misunderstood / misexplained. the z coord in `pts` variable which is being adjusted is the weather model z values. so that adjustment will always force it to the lowest z (-500; code needs to be updated to reflect this). There's a test in the last cells of [this notebook](https://github.com/dbekaert/RAiDER-docs/blob/66cd4e13dd55fd5ca3087b256abb1e4e7da780b8/notebooks/Synthetic_Test/SyntheticTest.ipynb).

Reply:
@bbuzz31 is correct in that this is I would say a very minor fix and doesn't have anything to do with the raytracing per se, has to do with the projections and numerical precision. I also think that this is needed to ensure the raytracing is tested correctly. The thing that needs to be done here is to ensure that the lowest z-level always gets set back to the input z-level rather than slightly beneath, as that results in nans.